### PR TITLE
Test cases added for gateway.go

### PR DESCRIPTION
**Pull request by shivi28**
**What type of PR is this?**
/area testing

**What this PR does / why we need it**:
Added test cases in gateway.go file

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2910 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Test coverage for gateway.go updated to 72%
```

### CODE REVIEW

File: pkg/cloud/services/network/gateways_test.go [hunk 1]
```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
...
+						{
+							Name:   aws.String("attachment.vpc-id"),
+							Values: aws.StringSlice([]string{"vpc-gateways"}),
+						},
+					},
+				})).Return(&ec2.DescribeInternetGatewaysOutput{}, nil)
+			},
+		},
+		{
+			name: "Should return without error if internet gateway is not found",
```
Comment (pydctw) [inline pkg/cloud/services/network/gateways_test.go:195]: I think you are testing a successful deletion of IGW. Could you change the test name to reflect it?

File: pkg/cloud/services/network/gateways_test.go [hunk 2]
```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
...
+	defer mockCtrl.Finish()
+
+	testCases := []struct {
+		name    string
+		input   *infrav1.NetworkSpec
+		expect  func(m *mock_ec2iface.MockEC2APIMockRecorder)
+		wantErr bool
+	}{
+		{
+			name: "Should return without error if vpc is unmanaged",
```
Comment (Ankitasw) [inline pkg/cloud/services/network/gateways_test.go:165]:
```suggestion
			name: "Should ignore deletion if vpc is unmanaged",
```

File: pkg/cloud/services/network/gateways_test.go [hunk 3]
```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
...
+			input: &infrav1.NetworkSpec{
+				VPC: infrav1.VPCSpec{
+					ID: "vpc-gateways",
+				},
+			},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {},
+		},
+		{
+			name: "Should return without error if internet gateway is not found",
```
Comment (Ankitasw) [inline pkg/cloud/services/network/gateways_test.go:174]:
```suggestion
			name: "Should ignore deletion if internet gateway is not found",
```

File: pkg/cloud/services/network/gateways_test.go [hunk 4]
```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
 		})
```
Comment (Ankitasw) [inline pkg/cloud/services/network/gateways_test.go:150]: Can we check the conditions set on AWSCluster in clusterscope object for success case?

Comment (shivi28) [reply pkg/cloud/services/network/gateways_test.go:150]: I think we could check this condition. But as per current PR scope, I have added the missing unit test cases and not focused on improving the existing test cases.
However to improve the existing test cases we could start a separate issue or can add it as part of adding integration tests(if required). Wdyt?

**Pull request merged**